src/_P023_OLED.ino

```diff
@@ -623,8 +623,13 @@ void Plugin_023_sendStrXY(struct Plugin_023_OLED_SettingStruct &oled,  const cha
   Plugin_023_setXY(oled, X, Y);
   unsigned char i = 0;
   unsigned char font_width = 0;
+  unsigned char currentPixels = Y * 8; // setXY always uses font_width = 8, Y = 0-based
```

Comment:
The fix is about the width of a line, so why initialize with `Y * 8` ?

Reply:
Setting the position using Plugin_023_setXY has a fixed font_width of 8 (pixels), as I wrote in the comment, and that should be accounted for to calculate the total width when not starting in column 1 (Y=0)

Reply:
Hmm, maybe I'm already a bit too tired, but I don't yet see why "column" and "Y" are related.
Also "font width" and Y are a bit conflicting in my head.
Could be that your code is correct, but then we really need to have a look at the variable naming as this looks really strange to me.

Reply:
Sorry, my explanation is probably a bit more confusing that it should be. I hope this clears things up...
For the `oled,x,y,text` command, the display is a matrix of 4-8 lines of 16 columns (assuming a 128 pixel wide display). That's why I referred to the Y argument in relation to the column.
For the Plugin_023_sendStrXY method, the display is both a matrix of 4-8 lines with 16 columns when it is using the X and Y arguments for positioning, but also a 128 pixel wide display when writing characters in an 8x8 matrix to the screen.

As the positioning part is based on that 8x8 matrix, the Y is multiplied by 8 to get the correct pixel offset. The variable that was already used for that width per character is `font_width`.
And then there is the Font Width: Optimized configuration setting... where the width of each character depends on the width is actually needs, from 8px for a W or a M, down to 5px for a !, an i or a period.
That's the reasoning for doing the pixel counting.

As said, it's tested using both Font Width settings, and no text spills over to the next line any longer. It is possible that characters are partially visible on the right edge of the display, but that is by intention.

Reply:
I still don't understand your approach.
It looks like your fix may only work if you print the lines from top to bottom.
But if you print just a single line update (not the last one), it looks like it is still wrapping to the next line.

It would make sense to me if you initialize the currentPixels value to something like `X * 8` and even better if you make it an int and initialize it to something like `64 - X*8` or `128 - X*8`. Then you only have to compare the pixels counter with 0.


For example something like this:
```c++
void Plugin_023_sendStrXY(struct Plugin_023_OLED_SettingStruct &oled,  const char *string, int X, int Y)
{
  Plugin_023_setXY(oled, X, Y);
  unsigned char i = 0;
  unsigned char font_width = 0;
  int pixelsLeft = 128;
  switch (oled.type) {
    case OLED_64x48:
    case OLED_64x48 | OLED_rotated:
	  pixelsLeft = 64;
      break;
  }
  pixelsLeft -= X * 8;

  while (*string && pixelsLeft > 0)
  {
    switch (oled.font_width)
    {
      case Size_optimized:
        font_width = pgm_read_byte(&(Plugin_023_myFont_Size[*string - 0x20]));
        break;
      default:
        font_width = 8;
    }

    for (i = 0; i < font_width && pixelsLeft > 0; i++)
    {
      Plugin_023_SendChar(oled, pgm_read_byte(Plugin_023_myFont[*string - 0x20] + i));
	  --pixelsLeft;
    }
    string++;
  }
}
```

Reply:
Well, the effect should be the same, except you used X, the row, instead of Y, the col(umn), to determine the offset.
And the OLED_rotated case that I didn't know about.
If it works, it works for me 😉

Reply:
I have not tested my suggested code.
But like I said, I really don't understand why the Y value should be used to determine the offset as you plan to limit the writing on a line of text.

Reply:
Just tested above code, on the display, using some system variables like ip, sunrise, sunset and systime (ntp is on), and all displayed text was wonky and overflowing, like you mention, until I replaced `pixelsLeft -= X * 8;` with `pixelsLeft -= Y * 8;`. Then it works exactly like my code, (I replaced my insufficient if with your switch/case for OLED_64x48 / OLED_rotated).
Now also tested configured for 64x48 display, and rotated, and no pixel leaks outside these boundaries when set.

Reply:
OK, so your code is working.
Only thing left is for me to know why it is working.

This may be the hardest part... overcome my stubbornness :)

Reply:
... or my inability to explain _how_ it is working 🤣

Reply:
Let me try one more time... 🤔 
The X value is the line you're writing, the Y value the position in default character width (8 px) to start at. They are passed to that method with 1 subtracted to become 0-based from the `oled,x,y,text` command, that is documented to have an upper-left position of 1,1 for x and y.
From that position in the line it writes each character from its 8x8 font matrix, 1 column of pixels at a time (the inner for loop). For Optimized font width, that matrix is between 5 and 8 pixels wide, after which the next character is started (leftmost pixel of each character is all 0's to have proper font spacing)

Reply:
Maybe I do have to look into the code of the library then, to see what is X and what is Y.
Feel a bit foggy in my head that I cannot wrap my head around it, probably mainly due to the labeling of the variables.
This does remind me of the time I spent to debug a motor control loop, where the variables for speed was called a and acceleration was called v (Dutch for "versnelling").
I just could not make sense of the (undocumented) code, just due to the strange naming.
I guess this is the same and if you would name it "foo" and "bar", I probably would understand it.

I will merge it as you stated it was working.